### DISCUSSION TITLE
appending trailing slash to canonical urls

### DISCUSSION THREAD

**Discussion by mrxsal** [vercel/next.js#53950] [App Router]
all my routes have trailing slashes. However with the following setup, the generated canonical URLs have no trailing slash:

```
export const generateMetadata = () => {
  return {
    metadataBase: new URL('https://example.com/'),
    alternates: {
      canonical: "./",
    }
}
```

also I have `trailingSlash: true` added in next.config.js

Is there a way I can achieve this programatically without having to set a canonical manually on every page?

**Comment by HellYeahOmg** [comment]
https://github.com/vercel/next.js/pull/62109 - check this pr
